schema-registry/src/main/java/io/streamnative/pulsar/handlers/kop/schemaregistry/model/impl/PulsarSchemaStorage.java

```diff
@@ -0,0 +1,579 @@
+                .blockIfQueueFull(true)
+                .createAsync();
+        return producerHandle.thenCompose(opProducer -> {
+            // nobody can write now to the topic
+            // wait for local cache to be up-to-date
+            CompletableFuture<List<T>> dummy = ensureLatestData(true)
+                    .thenCompose((___) -> {
+                        // build the Op, this will usually use the contents of the local cache
+                        List<Map.Entry<Op, T>> ops = opBuilder.get();
+                        List<T> res = new ArrayList<>();
+                        List<CompletableFuture<?>> sendHandles = new ArrayList<>();
+                        // write to Pulsar
+                        // if the write fails we lost the lock
+                        for (Map.Entry<Op, T> action : ops) {
+                            Op op = action.getKey();
+                            // if "op" is null, then we do not have to write to Pulsar
+                            if (op != null) {
+                                log.info("writing {} to Pulsar", op);
+                                if (!op.tenant.equals(getTenant())) {
+                                    sendHandles.add(FutureUtil.failedFuture(new SchemaStorageException(
+                                            "Invalid tenant " + op.tenant + ", expected " + tenant)));
+                                } else {
+                                    byte[] serialized = serializeOp(op);
+                                    String key = op.buildMessageKey();
+                                    sendHandles.add(opProducer
+                                            .newMessage()
+                                            .key(key)
+                                            .value(serialized)
+                                            .sendAsync().thenAccept((msgId) -> {
+                                        log.info("written {} as {} with key {} to Pulsar", op, msgId, key);
+                                        // write to local memory
+                                        applyOpToLocalMemory(op);
+                                    }));
+                                }
+                            }
+                            res.add(action.getValue());
+                        }
+
+                        return CompletableFuture
+                                .allOf(sendHandles.toArray(new CompletableFuture[0]))
+                                .thenApply(____ -> res);
+
+                    });
+            // ensure that we release the exclusive producer in any case
+            dummy.whenComplete((___, err) -> {
+                opProducer.closeAsync();
+            });
+            return dummy;
+        });
+
+
+    }
+
+    private byte[] serializeOp(Op op) {
+        try {
+            return MAPPER.writeValueAsBytes(op);
+        } catch (JsonProcessingException err) {
+            throw new RuntimeException(err);
```

Comment:
```suggestion
            throw new IllegalArgumentException(err);
```

[Codacy] Avoid throwing raw exception types.